# libcss: lexer
# 1 files, 17 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libcss/src/lex/lex.c browser/libcss/src/lex/lex.c
--- upstream/libcss/src/lex/lex.c	2026-09-20 23:27:15.833165409 +0100
+++ browser/libcss/src/lex/lex.c	2026-09-20 23:19:17.179471109 +0100
@@ -636,6 +636,23 @@
 			APPEND(lexer, cptr, clen);
 
 			t->type = CSS_TOKEN_CDC;
+		} else if (startNMStart(c)) {
+			/* "--" followed by nmstart is a CSS Custom Property
+			 * identifier. Keep both dashes in the token buffer
+			 * and continue consuming as an IDENT so the token
+			 * emitted is a single IDENT('--foo') rather than
+			 * CHAR('-') + IDENT('-foo'). */
+			if (c != '\\') {
+				APPEND(lexer, cptr, clen);
+				lexer->state = sIDENT;
+				lexer->substate = 0;
+				return IdentOrFunction(lexer, token);
+			} else {
+				lexer->bytesReadForToken += clen;
+				lexer->state = sIDENT;
+				lexer->substate = Escape;
+				goto escape;
+			}
 		} else {
 			/* Remove the '-' we read above */
 			lexer->bytesReadForToken -= 1;
